### PR TITLE
Add types generated from GraphQL schema

### PR DESCRIPTION
**Pull request by karolinakuzniewicz**

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3636 Setup codegen for schema types](https://app.clickup.com/t/2549495/SALEOR-3636)

File: codegen.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+overwrite: true
+schema: "https://master.staging.saleor.cloud/graphql/"
```
Comment (jwm0) [inline codegen.yml:2]: use env var for that

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -11,13 +11,14 @@
...
-    "build": "tsdx build",
+    "start": "npm run generate && tsdx watch",
+    "build": "npm run generate && tsdx build",
     "test": "tsdx test",
     "lint": "tsdx lint",
     "prepare": "tsdx build",
     "size": "size-limit",
-    "analyze": "size-limit --why"
+    "analyze": "size-limit --why",
+    "generate": "graphql-codegen --config codegen.yml"
```
Comment (krzysztofwolski) [inline package.json:21]: I would suggest same naming as in the dashboard 
https://github.com/mirumee/saleor-dashboard/blob/master/package.json#L231

File: codegen.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+generates:
+  src/apollo/types.ts:
+    plugins:
+      - "typescript"
+    config:
+      enumsAsTypes: true
+      skipTypename: true
```
Comment (krzysztofwolski) [inline codegen.yml:9]: @jwm0 should we skip typename? Dont remember why (caching in apollo?) we were using typenames in other projects.

File: codegen.yml (hunk 1 above)
Comment (krzysztofwolski) [reply codegen.yml:2]: Also, pointing to the staging api will result in random type changes on builds after core merges

File: codegen.yml (hunk 3 above)
Comment (krzysztofwolski) [reply codegen.yml:9]: I would also suggest adding `dedupeOperationSuffix` to avoid generating fragment types like `UserFragmentFragment`

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: codegen.yml (hunk 3 above)
Comment (karolinakuzniewicz) [reply codegen.yml:10]: done

**Review by krzysztofwolski — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**